### PR TITLE
docs: add aws-lambda CN document

### PR DESCRIPTION
**Pull request by penghuima**
Signed-off-by: penghuima <2992426202@qq.com>

### Description

add aws-lambda

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by juzhiyuan** [comment]
cc @SylviaBABY @hf400159, please have a review 🙏

**Comment by SylviaBABY** [comment]
> cc @SylviaBABY @hf400159, please have a review 🙏

Will review with #7094 together

File: docs/zh/latest/plugins/aws-lambda.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+`aws-lambda` 插件用于将 [AWS Lambda](https://aws.amazon.com/lambda/) 作为动态上游集成至 `APISIX` ，将所有特定 `URI` 的请求代理到 `AWS` 云。
+
+插件启用后会终止对已配置 `URI` 的请求，并代表客户端向 `AWS Lambda Gateway URI` 发起一个新的请求，其中包括配置的授权详细信息、请求头、正文（三个参数都是从原始请求中传递的）。然后将带有头信息、状态代码和正文的响应返回给使用 `APISIX` 发起请求的客户端。
+
+本插件支持通过 `AWS API key` 和 `AWS IAM secrets` 进行授权。
+
+## 属性
+
+| Name                 | Type    | Required | Default | Valid values | Description                                                  |
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:40]: Please translate, refer to [plugins/fault-injection.md](https://github.com/apache/apisix/blob/master/docs/zh/latest/plugins/fault-injection.md?plain=1).

File: docs/zh/latest/plugins/aws-lambda.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+| authorization.iam    | object  | 否       |         |              | 用于通过 `AWS v4` 请求签名执行的基于 AWS IAM 角色的授权。 请参阅 [IAM 授权方案](#IAM授权方案)。 |
+| timeout              | integer | 否       | 3000    | [100,...]    | 代理请求超时，以毫秒为单位。                                 |
+| ssl_verify           | boolean | 否       | true    | true/false   | 当设置为 `true` 时执行 `SSL` 验证。                          |
+| keepalive            | boolean | 否       | true    | true/false   | 当设置为 `true` 时，保持连接的活动状态以便重复使用。         |
+| keepalive_pool       | integer | 否       | 5       | [1,...]      | 在关闭该连接之前，可以在该连接上发送的最大请求数。           |
+| keepalive_timeout    | integer | 否       | 60000   | [1000,...]   | 在关闭该连接之前，可以在该连接上发送的最大请求数。           |
+
+### IAM 授权方案
+
+| Name       | Type   | Required | Default       | Description                                                  |
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:54]: Please translate, refer to [plugins/fault-injection.md](https://github.com/apache/apisix/blob/master/docs/zh/latest/plugins/fault-injection.md?plain=1).

File: docs/zh/latest/plugins/aws-lambda.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+                "apikey": "<Generated API Key from aws console>",
+            },
+            "ssl_verify":false
+        }
+    },
+    "uri": "/aws"
+}'
+```
+
+现在，任何对端点 `/aws` 的请求（`HTTP/1.1、HTTPS、HTTP2`）都将调用已配置的 `AWS` 函数的 `URI`，并且响应将被代理回给客户端。
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:81]:
```suggestion
现在，任何对端点 `/aws` 的请求（`HTTP/1.1`、`HTTPS`、`HTTP2`）都将调用已配置的 `AWS` 函数的 `URI`，并且响应将被代理回给客户端。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+x-amz-apigw-id: JdwXuEVxIAMFtKw=
+x-amzn-RequestId: 471289ab-d3b7-4819-9e1a-cb59cac611e0
+Content-Length: 16
+X-Amzn-Trace-Id: Root=1-61a22dca-600c552d1c05fec747fd6db0;Sampled=0
+Server: APISIX/2.10.2
+...
+"Hello, APISIX!"
+```
+
+另一个请求的例子是，客户端通过 `HTTP/2` 与 `APISIX` 进行通信，如下所示（请确保你在默认配置文件（`config-default.yaml`）中配置了 `enable_http2: true`。你可以通过取消对 `apisix.node_listen`  字段中端口`9081`的注释来做到这一点）：
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:103]:
```suggestion
另一个请求的例子是，客户端通过 `HTTP/2` 与 `APISIX` 进行通信，如下所示（请确保你在默认配置文件（`config-default.yaml`）中配置了 `enable_http2: true`。你可以通过取消对 `apisix.node_listen` 字段中端口 `9081` 的注释来做到这一点）：
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+                }
+            },
+            "ssl_verify": false
+        }
+    },
+    "uri": "/aws"
+}'
+```
+
+:::注意
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:143]:
```suggestion
:::note
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+这种方法假设你已经有一个 `IAM` 用户，并启用了程序化访问所需要的权限（`AmazonAPIGatewayInvokeFullAccess`）来访问端点。
+
+:::
+
+### 配置路由转发
+
+`aws-lambda` 插件在代理请求到 `AWS` 上游的时候也支持 `URL` 路径转发。基本请求路径的扩展被附加到插件配置中指定的 `function_uri` 上。
+
+:::重要信息
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:153]:
```suggestion
:::info IMPORTANT
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+在路由上配置的 `uri`  必须以 `*` 结尾，这样才能正常工作。`APISIX` 路由是严格匹配的，`*` 意味着这个 `URI` 的任何子路径都会被匹配到同一个路由。
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:155]:
```suggestion
在路由上配置的 `uri` 必须以 `*` 结尾，这样才能正常工作。`APISIX` 路由是严格匹配的，`*` 意味着这个 `URI` 的任何子路径都会被匹配到同一个路由。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+Content-Length: 16
+x-amz-apigw-id: JrHG8GC4IAMFaGA=
+Server: APISIX/2.11.0
+...
+"Hello, APISIX!"
+```
+
+## 禁用插件
+
+要禁用`aws-lambda`插件，你可以从插件配置中删除相应的 `JSON` 配置。你不需要重新启动就可以生效，因为 `APISIX` 将自动重新加载配置。
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:199]:
```suggestion
要禁用 `aws-lambda` 插件，你可以从插件配置中删除相应的 `JSON` 配置。你不需要重新启动就可以生效，因为 `APISIX` 将自动重新加载配置。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+            "ssl_verify":false
+        }
+    },
+    "uri": "/aws"
+}'
+```
+
+现在，任何对端点 `/aws` 的请求（`HTTP/1.1、HTTPS、HTTP2`）都将调用已配置的 `AWS` 函数的 `URI`，并且响应将被代理回给客户端。
+
+下面的例子中，`AWS Lambda` 从查询中接收名字，并返回一条 `"Hello $name"` 消息。
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:83]:
```suggestion
下面的例子中，`AWS Lambda` 从查询中接收名字，并返回一条 `"Hello $name"` 消息：
```

**Review by lunarwhite — COMMENTED** [review]
Thanks for your contribution!

File: docs/zh/latest/plugins/aws-lambda.md [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+x-amz-apigw-id: JdwXuEVxIAMFtKw=
+x-amzn-RequestId: 471289ab-d3b7-4819-9e1a-cb59cac611e0
+Content-Length: 16
+X-Amzn-Trace-Id: Root=1-61a22dca-600c552d1c05fec747fd6db0;Sampled=0
+Server: APISIX/2.10.2
+...
+"Hello, APISIX!"
+```
+
+另一个请求的例子是，客户端通过 `HTTP/2` 与 `APISIX` 进行通信，如下所示（请确保你在默认配置文件（`config-default.yaml`）中配置了 `enable_http2: true`。你可以通过取消对 `apisix.node_listen` 字段中端口 `9081` 的注释来做到这一点）：
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:103]:
```suggestion
下面是另一个请求的例子，客户端通过 HTTP/2 协议与 APISIX 进行通信。

在进行测试之前，请确保默认配置文件（`config-default.yaml`）中配置了 `enable_http2: true`。你可以通过取消对 `apisix.node_listen` 字段中端口 `9081` 的注释来配置此项。

使用 `curl` 命令测试：
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+
+```shell
+curl -i -XGET --http2 --http2-prior-knowledge localhost:9081/aws\?name=APISIX
+```
+
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:108]:
```suggestion

预期的返回结果：

```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+x-amz-apigw-id: JdwulHHrIAMFoFg=
+date: Sat, 27 Nov 2021 13:10:53 GMT
+x-amzn-trace-id: Root=1-61a22e5d-342eb64077dc9877644860dd;Sampled=0
+x-amzn-requestid: a2c2b799-ecc6-44ec-b586-38c0e3b11fe4
+server: APISIX/2.10.2
+...
+"Hello, APISIX!"
+```
+
+同样，该函数也可以通过 `AWS API Gateway` 使用 `AWS IAM` 权限进行授权而触发。该插件通过 `AWS v4` 请求签名在 `HTTP` 调用中包含了身份验证签名。下面的例子显示了此方法。
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:122]:
```suggestion
与上面的示例类似，AWS Lambda 函数也可以通过 AWS API Gateway 触发，但需要使用 AWS IAM 权限进行授权。`aws-lambda` 插件的配置文件中包含了 `"authorization"` 字段，用户可以在 HTTP 调用中通过 AWS v4 请求签名。

以下示例展示了如何通过配置文件实现授权：
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+            "ssl_verify": false
+        }
+    },
+    "uri": "/aws"
+}'
+```
+
+:::note
+
+这种方法假设你已经有一个 `IAM` 用户，并启用了程序化访问所需要的权限（`AmazonAPIGatewayInvokeFullAccess`）来访问端点。
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:145]:
```suggestion
此方法假设你已经有一个启用了程序化访问的 IAM 用户，并具有访问端点的必要权限（AmazonAPIGatewayInvokeFullAccess）。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+
+:::
+
+### 配置路径转发
+
+`aws-lambda` 插件在代理请求到 AWS 上游的时候也支持 URL 路径转发。基本请求路径的扩展被附加到插件配置中指定的 `function_uri` 字段上。
+
+:::info IMPORTANT
+
+为了使 `aws-lambda` 插件正常工作，在路由上配置的 `uri` 字段必须以 `*` 结尾。这是因为 APISIX 路由是严格匹配的，`*` 意味着这个 URI 的任何子路径都会被匹配到同一个路由。
```
Comment (hf400159) [inline docs/zh/latest/plugins/aws-lambda.md:165]:
```suggestion
因为 APISIX 路由是严格匹配的，所以为了使 `aws-lambda` 插件正常工作，在路由上配置的 `uri` 字段必须以 `*` 结尾，`*` 意味着这个 URI 的任何子路径都会被匹配到同一个路由。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+
+:::note
+
+此方法假设你已经有一个启用了程序化访问的 IAM 用户，并具有访问端点的必要权限（AmazonAPIGatewayInvokeFullAccess）。
+
+:::
+
+### 配置路径转发
+
+`aws-lambda` 插件在代理请求到 AWS 上游的时候也支持 URL 路径转发。基本请求路径的扩展被附加到插件配置中指定的 `function_uri` 字段上。
```
Comment (hf400159) [inline docs/zh/latest/plugins/aws-lambda.md:161]:
```suggestion
`aws-lambda` 插件在代理请求到 AWS 上游时也支持 URL 路径转发。基本请求路径的扩展被附加到插件配置中指定的 `function_uri` 字段上。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+                "apikey": "<Generate API key>"
+            },
+            "ssl_verify":false
+        }
+    },
+    "uri": "/aws/*"
+}'
+```
+
+通过上述示例配置插件后，任何对 `aws/default/test-apisix` 路径的请求都会调用 AWS Lambda 函数，并转发添加的路径。
```
Comment (hf400159) [inline docs/zh/latest/plugins/aws-lambda.md:187]:
```suggestion
通过上述示例配置插件后，任何访问 `aws/default/test-apisix` 的请求都会调用 AWS Lambda 函数，并转发附加的参数。
```

Comment (hf400159) [reply docs/zh/latest/plugins/aws-lambda.md:187]: I'm not sure if that description is correct.

**Review by hf400159 — COMMENTED** [review]
LGTM

**Review by SylviaBABY — DISMISSED** [review]

**Comment by SylviaBABY** [comment]
@penghuima Thanks for your contribution!

@lunarwhite And glad to see you as Reviewer, great work^^

File: docs/zh/latest/plugins/aws-lambda.md [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+| function_uri         | string  | 是       |         |              | 触发 lambda serverless 函数的 AWS API Gateway 端点。        |
+| authorization        | object  | 否       |         |              | 访问云函数的授权凭证。                                       |
+| authorization.apikey | string  | 否       |         |              | 生成的 API 密钥，用于授权对 AWS Gateway 端点的请求。         |
+| authorization.iam    | object  | 否       |         |              | 用于通过 AWS v4 请求签名执行的基于 AWS IAM 角色的授权。 请参考 [IAM 授权方案](#IAM授权方案)。 |
+| timeout              | integer | 否       | 3000    | [100,...]    | 代理请求超时（以毫秒为单位）。                                 |
+| ssl_verify           | boolean | 否       | true    | true/false   | 当设置为 `true` 时执行 SSL 验证。                          |
+| keepalive            | boolean | 否       | true    | true/false   | 当设置为 `true` 时，保持连接的活动状态以便重复使用。         |
+| keepalive_pool       | integer | 否       | 5       | [1,...]      | 在关闭该连接之前，可以在该连接上发送的最大请求数。           |
+| keepalive_timeout    | integer | 否       | 60000   | [1000,...]   | 在关闭该连接之前，可以在该连接上发送的最大请求数。           |
```
Comment (hf400159) [inline docs/zh/latest/plugins/aws-lambda.md:50]:
```suggestion
| keepalive_timeout    | integer | 否       | 60000   | [1000,...]   | 当连接空闲时，保持该连接处于活动状态的时间（以毫秒为单位）。           |
```
```suggestion
| keepalive_timeout    | integer | 否       | 60000   | [1000,...]   | 在关闭该连接之前，可以在该连接上发送的最大请求数。           |
```

Comment (hf400159) [reply docs/zh/latest/plugins/aws-lambda.md:50]: 请同步修改英文。

File: docs/zh/latest/plugins/aws-lambda.md [hunk 18]
```diff
@@ -0,0 +1,217 @@
...
+
+### IAM 授权方案
+
+| 名称       | 类型    | 必选项   | 默认值         | 描述                                                  |
+| ---------- | ------ | -------- | ------------- | ------------------------------------------------------------ |
+| accesskey  | string | 是       |               | 从 AWS IAM 控制台生成的访问密钥 ID。                     |
+| secret_key | string | 是       |               | 从 AWS IAM 控制台生成的访问密钥。                          |
+| aws_region | string | 否       | "us-east-1"   | 发出请求的 AWS 区域。                                    |
+| service    | string | 否       | "execute-api" | 接收该请求的服务。特别的，对于 HTTP 触发器是 `"execute-api"`。 |
```
Comment (hf400159) [inline docs/zh/latest/plugins/aws-lambda.md:59]:
```suggestion
| service    | string | 否       | "execute-api" | 接收该请求的服务。请注意，对于 HTTP 触发器是 `"execute-api"`。 |
```

**Review by hf400159 — APPROVED** [review]

**Review by SylviaBABY — DISMISSED** [review]

**Review by bzp2010 — DISMISSED** [review]

File: docs/zh/latest/plugins/aws-lambda.md [hunk 19]
```diff
@@ -0,0 +1,217 @@
+---
+title: aws-lambda
+keywords:
+  - APISIX
+  - Plugin
+  - AWS Lambda
+  - aws-lambda
+description: 本文介绍了关于 Apache APISIX `aws-lambda` 插件的基本信息及使用方法。
```
Comment (juzhiyuan) [inline docs/zh/latest/plugins/aws-lambda.md:8]:
```suggestion
description: 本文介绍了关于 Apache APISIX aws-lambda 插件的基本信息及使用方法。
```

File: docs/zh/latest/plugins/aws-lambda.md [hunk 20]
```diff
@@ -0,0 +1,217 @@
...
+| keepalive_pool       | integer | 否       | 5       | [1,...]      | 在关闭该连接之前，可以在该连接上发送的最大请求数。           |
+| keepalive_timeout    | integer | 否       | 60000   | [1000,...]   | 当连接空闲时，保持该连接处于活动状态的时间（以毫秒为单位）。           |
+
+### IAM 授权方案
+
+| 名称       | 类型    | 必选项   | 默认值         | 描述                                                  |
+| ---------- | ------ | -------- | ------------- | ------------------------------------------------------------ |
+| accesskey  | string | 是       |               | 从 AWS IAM 控制台生成的访问密钥 ID。                     |
+| secret_key | string | 是       |               | 从 AWS IAM 控制台生成的访问密钥。                          |
+| aws_region | string | 否       | "us-east-1"   | 发出请求的 AWS 区域。                                    |
```
Comment (juzhiyuan) [inline docs/zh/latest/plugins/aws-lambda.md:58]: Do we have a region list?

File: docs/zh/latest/plugins/aws-lambda.md [hunk 21]
```diff
@@ -0,0 +1,217 @@
...
+| service    | string | 否       | "execute-api" | 接收该请求的服务。请注意，对于 HTTP 触发器是 `"execute-api"`。 |
```
Comment (juzhiyuan) [inline docs/zh/latest/plugins/aws-lambda.md:59]: Could we link a doc to describe this field?

**Review by juzhiyuan — DISMISSED** [review]

File: docs/zh/latest/plugins/aws-lambda.md (hunk 21 above)
Comment (penghuima) [reply docs/zh/latest/plugins/aws-lambda.md:59]: > Could we link a doc to describe this field?

like this  [aws_region](https://docs.aws.amazon.com/zh_cn/general/latest/gr/rande.html#region-names-codes)

Comment (juzhiyuan) [reply docs/zh/latest/plugins/aws-lambda.md:59]: Yes, then users will know what fields AWS support :)

File: docs/zh/latest/plugins/aws-lambda.md (hunk 20 above)
Comment (juzhiyuan) [reply docs/zh/latest/plugins/aws-lambda.md:58]: Yes, then users will know what fields AWS support :)

cc @penghuima

**Review by SylviaBABY — DISMISSED** [review]

**Review by hf400159 — DISMISSED** [review]

File: docs/zh/latest/plugins/aws-lambda.md [hunk 22]
```diff
@@ -0,0 +1,217 @@
...
+该插件支持通过 AWS API key 和 AWS IAM secrets 进行授权。
+
+## 属性
+
+| 名称                   | 类型     | 必选项 | 默认值  | 有效值       | 描述                                                 |
+| ------------------- | ------- | ------ | ------- | ------------ | ------------------------------------------------------------ |
+| function_uri         | string  | 是       |         |              | 触发 lambda serverless 函数的 AWS API Gateway 端点。        |
+| authorization        | object  | 否       |         |              | 访问云函数的授权凭证。                                       |
+| authorization.apikey | string  | 否       |         |              | 生成的 API 密钥，用于授权对 AWS Gateway 端点的请求。         |
+| authorization.iam    | object  | 否       |         |              | 用于通过 AWS v4 请求签名执行的基于 AWS IAM 角色的授权。 请参考 [IAM 授权方案](#IAM授权方案)。 |
```
Comment (lunarwhite) [inline docs/zh/latest/plugins/aws-lambda.md:45]:
```suggestion
| authorization.iam    | object  | 否       |         |              | 用于通过 AWS v4 请求签名执行的基于 AWS IAM 角色的授权。 请参考 [IAM 授权方案](#IAM-授权方案)。 |
```

**Review by lunarwhite — APPROVED** [review]
LGTM, just a tiny catch.

**Comment by spacewander** [comment]
Let's merge master to make CI pass.

**Comment by spacewander** [comment]
@juzhiyuan 
Please take a look. Thanks!

**Review by SylviaBABY — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**